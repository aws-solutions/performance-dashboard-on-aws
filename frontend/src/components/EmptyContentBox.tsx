import React from "react";

interface Props {
  onClick?: Function;
}

function EmptyContentBox(props: Props) {
  return (
    <div className="text-center radius-lg padding-5 margin-y-3 border-base border-dashed bg-base-lightest border">
      <p>
        This dashboard is empty. Build the dashboard by adding <br />
        charts, tables, metrics and text as content.
      </p>
      <button
        className="usa-button usa-button--base margin-top-1"
        onClick={() => {
          if (props.onClick) {
            props.onClick();
          }
        }}
      >
        + Add content
      </button>
    </div>
  );
}

export default EmptyContentBox;
