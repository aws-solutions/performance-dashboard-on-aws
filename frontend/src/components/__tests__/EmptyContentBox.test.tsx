import React from "react";
import { render, fireEvent } from "@testing-library/react";
import EmptyContentBox from "../EmptyContentBox";

describe("EmptyContentBox", () => {
  test("renders an empty box", async () => {
    const onClick = jest.fn();
    const wrapper = render(<EmptyContentBox widgets={[]} onClick={onClick}/>);
    expect(wrapper.container).toMatchSnapshot();
  });

  test("calls the onClick callback when button is pressed", async () => {
    const onClick = jest.fn();
    const wrapper = render(<EmptyContentBox widgets={[]} onClick={onClick} />);
    fireEvent.click(wrapper.getByRole("button"));
    expect(onClick).toBeCalled();
  });
});