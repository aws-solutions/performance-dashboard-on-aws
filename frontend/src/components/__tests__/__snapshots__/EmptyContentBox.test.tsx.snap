// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyContentBox renders an empty box 1`] = `
<div>
  <div>
    <div
      class="text-center radius-lg padding-5 margin-y-3 border-base border-dashed bg-base-lightest border"
    >
      <p>
        This dashboard is empty. Build the dashboard by adding 
        <br />
        charts, tables, metrics and text as content.
      </p>
    </div>
    <div
      class="text-center margin-top-2"
    >
      <button
        class="usa-button usa-button--base margin-top-1"
      >
        + Add content
      </button>
    </div>
  </div>
</div>
`;
